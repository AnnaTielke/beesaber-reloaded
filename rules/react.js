module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
