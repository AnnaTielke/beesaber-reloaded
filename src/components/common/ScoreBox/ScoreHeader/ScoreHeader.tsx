import { MDBInput } from 'mdb-react-ui-kit';
import React from 'react';
import styled from 'styled-components';

import Title from '@/components/common/Title/Title';
import { useAppSelector } from '@/store/hooks';
import { selectUserName } from '@/store/reducer/userDataReducer';
import tokens from '@/tokens';

const Header = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  margin: 0;
  color: ${tokens.color.blue.light};
  label {
    color: ${tokens.color.white.light}!important;
  }
`;

type Props = {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  query: string;
};

const ScoreHeader = ({ onChange, query }: Props): JSX.Element => {
  const username = useAppSelector(selectUserName);
  const title = username !== undefined ? username.toUpperCase() : 'MY SCORES';

  return (
    <Header>
      <Title as="h3">{title}</Title>
      <MDBInput
        label="Filter Songs..."
        size="sm"
        value={query}
        onChange={onChange}
      />
    </Header>
  );
};

export default ScoreHeader;
