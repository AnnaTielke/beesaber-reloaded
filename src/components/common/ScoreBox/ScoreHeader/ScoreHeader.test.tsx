import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import ScoreHeader from './ScoreHeader';

const mockStore = configureMockStore();

describe('ScoreHeader', () => {
  const mockOnChange = jest.fn();

  it.each(['MyUserName', undefined])(
    'should match the snapshot',
    (username) => {
      const store = {
        userData: {
          username,
        },
      };

      const { container } = render(
        <Provider store={mockStore(store)}>
          <ScoreHeader onChange={mockOnChange} query={'some query'} />
        </Provider>
      );

      expect(container.firstChild).toMatchSnapshot();
    }
  );

  it('should call onChange when typing a query', () => {
    const store = {
      userData: {
        username: 'MyUserName',
      },
    };
    const someQuery = 'word';

    render(
      <Provider store={mockStore(store)}>
        <ScoreHeader onChange={mockOnChange} query={'some query'} />
      </Provider>
    );

    const input = screen.getByRole('textbox');

    userEvent.type(input, someQuery);

    expect(mockOnChange).toBeCalledTimes(4);
  });
});
