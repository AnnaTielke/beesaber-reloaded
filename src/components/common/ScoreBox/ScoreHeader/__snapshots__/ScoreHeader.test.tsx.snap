// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreHeader should match the snapshot 1`] = `
.c2 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin: 0;
  color: #edf5ff;
}

.c0 label {
  color: rgba(255,255,255,0.7)!important;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      MYUSERNAME
    </h3>
  </div>
  <div
    class="form-outline"
  >
    <input
      class="form-control active form-control-sm"
      value="some query"
    />
    <label
      class="form-label"
    >
      Filter Songs...
    </label>
    <div
      class="form-notch"
    >
      <div
        class="form-notch-leading"
      />
      <div
        class="form-notch-middle"
        style="width: 0px;"
      />
      <div
        class="form-notch-trailing"
      />
    </div>
  </div>
</div>
`;

exports[`ScoreHeader should match the snapshot 2`] = `
.c2 {
  color: #fff0f0;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #ff0025 -2px -1px 5px,#ff0025 2px 1px 5px, #ff0025 0px 0px 10px,#ff0025 0px 0px 15px,#ff0025 0px 0px 20px, #ff0025 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c3 {
  color: #edf5ff;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -moz-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -o-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  -webkit-transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  transition: color 0.6s ease-out,text-shadow 0.6s ease-out;
  text-shadow: #0000FF -2px -1px 5px,#0000FF 2px 1px 5px, #0000FF 0px 0px 10px,#0000FF 0px 0px 15px,#0000FF 0px 0px 20px, #0000FF 0px 0px 30px;
  font-family: 'NeonTubes2';
  font-weight: 300;
  margin: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin: 0;
  color: #edf5ff;
}

.c0 label {
  color: rgba(255,255,255,0.7)!important;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3
      class="c2"
    >
      MY
    </h3>
    <h3
      class="c3"
    >
      SCORES
    </h3>
  </div>
  <div
    class="form-outline"
  >
    <input
      class="form-control active form-control-sm"
      value="some query"
    />
    <label
      class="form-label"
    >
      Filter Songs...
    </label>
    <div
      class="form-notch"
    >
      <div
        class="form-notch-leading"
      />
      <div
        class="form-notch-middle"
        style="width: 0px;"
      />
      <div
        class="form-notch-trailing"
      />
    </div>
  </div>
</div>
`;
